src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -699,10 +699,22 @@ private static bool CanReplace(ISymbol symbol)
                 return false;
             }
 
-            if (optionSet.GetOption(SimplificationOptions.QualifyMemberAccessWithThisOrMe, semanticModel.Language) &&
-                memberAccess.Expression.Kind() == SyntaxKind.ThisExpression)
+            if (memberAccess.Expression.IsKind(SyntaxKind.ThisExpression))
             {
-                return false;
+                // if we can't resolve the symbol then nothing can be inferred
+                var nameSymbol = SimplificationHelpers.GetOriginalSymbolInfo(semanticModel, memberAccess.Name);
+                if (nameSymbol == null)
+                {
+                    return false;
+                }
+
+                if ((nameSymbol.IsKind(SymbolKind.Field) && optionSet.GetOption(SimplificationOptions.QualifyMemberFieldAccessWithThisOrMe, semanticModel.Language)) ||
```

Comment:
This entire if statement seems to be repeated below.  Can you have a helper for this?